raft/node.go

```diff
@@ -427,6 +438,15 @@ func (n *node) ReportUnreachable(id uint64) {
 	}
 }
 
+func (n *node) ReportSnapshot(id uint64, status SnapshotStatus) {
```

Comment:
I have a little concern about outdated snapshot report. If external world reports outdated snapshot report, it may trigger the resend logic. But this should be a low-probability case.